### DISCUSSION TITLE
Offline no-position-lock stepping through turn-by-turn navigation

### DISCUSSION THREAD

**Discussion by dileepvr** [rinigus/pure-maps#620] [Ideas]
This is a useful feature for offline use cases (for privacy) without a gps lock (due to bad signal or lack of hardware receiver). Puremaps can search and route offline using the OSMscoutserver or equivalent.

It shouldn't be hard to code a mode that steps through (forward or backward) a route's directions either through manual button presses, or voice commands (if running on a device with a mic). Both display and audio prompts can respond as if the device's position as triggered an automatic step-through the turns in the route.

Using the GUI, the user is free to exit the mode, change the route by dragging, and re-enter the route if a wrong turn is made in real space.

**Comment by jonnius** [comment]
You can use the developer mode in the settings to make Pure Maps assume your location at the map center. Then you can move along the route by moving the map.

**Comment by rinigus** [comment]
I agree with @jonnius, developer mode can be used for that. Although, it is not as user friendly for this particular scenario. However, I wouldn't see myself working on it as there are many other issues in front of it for me in terms of priority. So, current developer mode is probably the best you can expect for a long while unless there is someone wishing to work on it.
